### ISSUE TITLE
Images cannot be parsed when obtained through event handlers

### ISSUE THREAD

**Issue by GalacticHypernova** [nuxt/image#1329]
Currently NuxtImg only seems to work fine with images that are presented at build time in `~/public`. While that may suffice most static sites, for sites that require user interaction, including uploading media, this ends up breaking the images in production. It is not ideal to have to rebuild the app every time an image is uploaded, for quite obvious reasons.
Take for example this code:
```ts

// ~/server/routes/images/[image].get.ts
export default defineEventHandler({
    handler: async (event) => {
        const img = getRouterParam(event, "image")
        const imageStorage = useStorage("images") // this could, for example, be an FS driver configured in `nuxt.config.ts`
        if (await imageStorage.hasItem(img)) {
            return await imageStorage.getItem(img)
        }
    }
})
```
This works fine when going to it on the browser, but NuxtImg fails to parse this, with a message stating the file cannot be found.

It would be very beneficial to have the benefits of image optimization for images that are obtained dynamically through event handlers, as most SSR apps use this kind of functionality.

**Comment by GalacticHypernova** [comment]
Hey @Baroshem ! I added the reproduction of the event handler that NuxtImg can't pick up. Daniel also stated that it is currently a limitation of NuxtImg, that it can only reach images present in `~/public` (or those from external services). Is there any part of the reproduction that you want specifically or would you like just a full minimal project to demonstrate it? Because the event handler I sent here can be real easily attached to a project and then referenced in the src of the image, which will lead to the error.

**Comment by GalacticHypernova** [comment]
Hey again @Baroshem !

I made some more digging, and I think I understand why this is happening.

As long as you don't outright disable FS storage, Nuxt automatically registers IPX FS:
https://github.com/nuxt/image/blob/27dbb2e36edbb11ce11cc606a09ca3ca92d10e4e/src/ipx.ts#L46C5-L49C7


IPX, in response, uses the FS storage when you provide an image, regardless of whether it's actually a public asset or something fetched on the server-side:
https://github.com/unjs/ipx/blob/main/src/ipx.ts#L102C4-L104C48

So if you have a route named `/images/[image].get.ts` (this might even be happening with direct placement of the image file there, though I haven't tested), any call to it would get treated as an FS call to /public/images, thus failing to parse images from a server route.

**Comment by Baroshem** [comment]
Thanks @GalacticHypernova for the additional digging and concept!

@danielroe what do you think about it? Do we want to support image optimization for images that are obtained dynamically through event handlers, as explained by GalacticHypernova in the initial feature request?

**Comment by danielroe** [comment]
yes we do. there's an open PR to do it, which needs a few fixes.

**Comment by Baroshem** [comment]
Could you link the open PR here so that we can keep track of it? :)
